src/main/java/State/State.java

```diff
@@ -0,0 +1,92 @@
+package State;
+
+import java.util.ArrayList;
+
+public class State implements Cloneable {
+    // 0 = empty, 1 = black queen, 2 = white queen, 3 = arrow
+    private byte[][] board;
+    private int[][] blackQueens;
+    private int[][] whiteQueens;
+
+    public final int BOARD_SIZE = 10;
+    public static final int WHITE_QUEEN = 2;
+    public static final int BLACK_QUEEN = 1;
+    public static final int ARROW = 3;
+
+    public State(ArrayList<Integer> gameState) {
+        board = new byte[BOARD_SIZE][BOARD_SIZE];
+        blackQueens = new int[4][2];
+        whiteQueens = new int[4][2];
+
+        int blackQueensFound = 0;
+        int whiteQueensFound = 0;
+        for (int x = 0; x < BOARD_SIZE; x++) {
+            for (int y = 0; y < BOARD_SIZE; y++) {
+                byte piece = (byte)(int)gameState.get((y+1)*11 + x+1);
+                board[x][y] = piece;
+
+                // Save where the queens are placed
+                if (piece == BLACK_QUEEN) {
+                    blackQueens[blackQueensFound++] = new int[]{x,y};
+                } else if (piece == WHITE_QUEEN) {
+                    whiteQueens[whiteQueensFound++] = new int[]{x,y};
+                }
+            }
+        }
+    }
+
+    public State(State state, Action action) {
+        try {
+            State cloned = (State) state.clone();
+            board = cloned.board;
+            board[action.getNewPos().get(0)][action.getNewPos().get(1)] = board[action.getOldPos().get(0)][action.getOldPos().get(1)];
```

Comment:
Did we want to think of a way to track which player we actually are? We might want another class initialized when the game starts so that we can store which colour we are and use the everywhere instead of this.